Library/Homebrew/api.rb

```diff
@@ -58,5 +55,18 @@ def fetch_json_api_file(endpoint, target:)
         retry
       end
     end
+
+    sig { params(token: String, git_head: T.nilable(String)).returns(String) }
+    def fetch_source(token, git_head: nil)
+      git_head ||= "master"
+      endpoint = "#{git_head}/Casks/#{token}.rb"
+      return cache[endpoint] if cache.present? && cache.key?(endpoint)
+
+      raw_url = "https://raw.githubusercontent.com/Homebrew/homebrew-cask/#{endpoint}"
+      output = Utils::Curl.curl_output("--fail", raw_url, max_time: 5)
+      raise ArgumentError, "No file found at #{Tty.underline}#{raw_url}#{Tty.reset}" unless output.success?
+
+      cache[endpoint] = output.stdout
+    end
```

Comment:
I think this method should either be generalized to formulae and casks (i.e. passing a repo and filepath instead of just a token) or be moved to `Homebrew::API::Cask`. This file contains the more general API things that might be used by multiple API types.